### PR TITLE
Refactor: rename name attributes to filename

### PR DESCRIPTION
**Pull request by alexanderleegs**
This PR accomplishes 2 things:

- Renaming `name` attributes of collection and resource pages to `fileName`
- Updated Proptypes for `CollectionPagesSection`, `FolderCreationModal`, and `FolderContent`

Currently, we have a discrepancy between the processing of filenames between collection and resource pages - we have been processing some items as name attributes and some as fileName. This PR attempts to standardise this naming convention to use `filename`.

### CODE REVIEW

**Review by gweiying — APPROVED** [review]
lgtm!

File: src/components/CollectionPagesSection.jsx [hunk 1]
```diff
@@ -210,7 +210,7 @@ const CollectionPagesSection = ({ collectionName, pages, siteName, isResource, r
                                     itemIndex={pageIdx}
                                     category={collectionName}
                                     siteName={siteName}
-                                    fileName={page.fileName || page.name} // temporary fix
+                                    fileName={page.fileName} // temporary fix
```
Comment (gweiying) [inline src/components/CollectionPagesSection.jsx:213]: nit: can remove comment now

File: src/components/FolderCreationModal.jsx [hunk 2]
```diff
@@ -30,6 +30,7 @@ const FolderCreationModal = ({
   siteName,
   setIsFolderCreationActive,
 }) => {
+  console.log(pagesData)
```
Comment (gweiying) [inline src/components/FolderCreationModal.jsx:33]: nit: remove log statement

File: src/components/CollectionPagesSection.jsx (hunk 1 above)
Comment (alexanderleegs) [reply src/components/CollectionPagesSection.jsx]: Resolved in https://github.com/isomerpages/isomercms-frontend/pull/412/commits/1ce7153a42365a2ccb6faa68be0fa264fb721d53

File: src/components/FolderCreationModal.jsx (hunk 2 above)
Comment (alexanderleegs) [reply src/components/FolderCreationModal.jsx]: Resolved in https://github.com/isomerpages/isomercms-frontend/pull/412/commits/f54ce6788955e338947448a8ef1423e7cb07e919

**Review by gweiying — APPROVED** [review]
LGTM!

**Pull request merged**